### DISCUSSION TITLE
incorrect format of week

### DISCUSSION THREAD

**Discussion by wenqi73** [js-fns/date-fns#1818] [General] [answered]
Hi, I use `yyyy-ww` to format date, but the return week is `01`.
```ts
import { format } from 'date-fns';
format(new Date(2019, 11, 30), 'yyyy-ww');
// 2019-01
```

**Comment by SoniaLahi** [comment, marked as answer]
This behavior is correct. Indeed, Monday 30th December 2019 is a Monday, the first Monday of the first week of 2020.

You can see more here : http://week-number.net/calendar-with-week-numbers-2019.html

I have tried with these values : 
```javascript
new Date(2019, 11, 28)  // Saturday  -> output 52
new Date(2019, 11, 29) // Sunday -> output 52 if first day of week is set to Monday, otherwise by default (Sunday) output is 01
new Date(2019, 11, 30) // Monday -> output 01
new Date(2019, 11, 31) // Tuesday -> output 01
```

**Comment by wenqi73** [comment]
@kossnocorp Could you please check this issue?

**Comment by kossnocorp** [comment]
@wenqi73 @SoniaLahi did answer your question (thanks, Sonia!). You maybe want to use ISO-week numbering `II` instead. Or not. It highly depends on your use-case. I recommend you to read about it more to figure out what you actually need (you can start here: https://en.wikipedia.org/wiki/Week#Week_numbering).

If you actually need `ww`, you must use `YYYY-ww` which would produce (with `II` you must use `RRRR`):

```
format(new Date(2019, 11, 30), 'YYYY-ww');
//=> 2020-01
```

**Reply by wenqi73** [reply]
@kossnocorp Thanks for your explanation! I have a little confusing, the usage of `ww` is similar to ISO-week, but it regards Sunday as the beginning of the week(ISO-week uses Monday as the beginning), seems it's not a standard format.

**Reply by shawncao** [reply]
I see this error:

Use `yyyy` instead of `YYYY` (in `YYYY-ww`) for formatting years to the input `1735459200000`

Looks like YYYY is not allowed in formatting week.

I have the same issue - format(date(2019-12-29)), `'W'ww 'Y'yy`) => I expect `W1 Y20` not `W1 Y19`, any solution for this?
